lib/ParseSIL/ParseSIL.cpp

```diff
@@ -5723,19 +5734,7 @@ bool SILParserTUState::parseDeclSIL(Parser &P) {
 
       // SWIFT_ENABLE_TENSORFLOW
       for (auto &attr : DiffAttrs) {
-        // Resolve parameter indices to have the right capacity, if it's
```

Comment:
Basically: the "update parameter indices capacity" logic needs to be run even when the function has no body.
The "resolve where clause requirements" logic cannot be run when the function has no body, because generic environment (necessary for resolution) is set only when the body exists.